src/Block/BlockContextManager.php

```diff
@@ -106,8 +105,12 @@ public function get($meta, array $settings = []): BlockContextInterface
                 $e->getMessage()
             ));
 
-            // NEXT_MAJOR: Only pass the template value if it's a string.
-            $settings = $this->resolve($block, $settings + ['template' => $block->getSetting('template')]);
+            $template = $block->getSetting('template');
+
+            $settings = $this->resolve(
+                $block,
+                $settings + (\is_string($template) ? ['template' => $template] : [])
```

Comment:
is this what was meant by `NEXT_MAJOR: Only pass the template value if it's a string.` @VincentLanglet ?